builtin/providers/ibmcloud/data_source_ibmcloud_infra_ssh_key.go

```diff
@@ -0,0 +1,110 @@
+package ibmcloud
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+)
+
+func dataSourceIBMCloudInfraSSHKey() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMCloudInfraSSHKeyRead,
+
+		Schema: map[string]*schema.Schema{
+			"label": &schema.Schema{
+				Description: "The label associated with the ssh key",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"public_key": &schema.Schema{
+				Description: "The public ssh key",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"fingerprint": &schema.Schema{
+				Description: "A sequence of bytes to authenticate or lookup a longer ssh key",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"notes": &schema.Schema{
+				Description: "A small note about a ssh key to use at your discretion",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"most_recent": &schema.Schema{
+				Description: "If true and multiple entries are found, the most recently created key is used. " +
+					"If false, an error is returned",
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func dataSourceIBMCloudInfraSSHKeyRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	service := services.GetAccountService(sess)
+
+	label := d.Get("label").(string)
+	mostRecent := d.Get("most_recent").(bool)
+
+	keys, err := service.
+		Filter(filter.Build(filter.Path("sshKeys.label").Eq(label))).
+		Mask("id,label,key,fingerprint,notes,createDate").
```

Comment:
We actually don't need to provide them, but doing so saves some data transfer over the wire.
It is well supported at the API server itself and hence we are leveraging that. In this case we might not save much but in other cases when payload may be huge it helps us bring only what we need from the server